xds/internal/client/client_xds.go

```diff
@@ -417,6 +421,9 @@ func securityConfigFromCluster(cluster *v3clusterpb.Cluster) (*SecurityConfig, e
 	if ts == nil {
 		return nil, nil
 	}
+	if name := ts.GetName(); name != transportSocketName {
```

Comment:
Do we know what other potential value this could be?

Can it be "insecure"? Like `tcp`?
I assume they wouldn't set this field at all for "insecure"?
But the field is only called `TransportSocket`. What if it's set in the insecure cases?

Reply:
For insecure, it looks like this field is not populated at all:
```
  // If no transport socket configuration is specified, new connections
  // will be set up with plaintext.
```

Reply:
> Do we know what other potential value this could be?

https://github.com/envoyproxy/envoy/blob/master/source/extensions/transport_sockets/well_known_names.h#L15

But it does look like TD is expected to send only the value that we are checking against.